### PR TITLE
[mono] Remove verifier

### PR DESCRIPTION
**Pull request by lambdageek**
There are a 3 MONO_API functions that remain.

### CODE REVIEW

**Review by CoffeeFlux — APPROVED** [review]

File: src/mono/mono/mini/debugger-agent.c [hunk 1]
```diff
@@ -8511,8 +8510,6 @@ method_commands_internal (int command, MonoMethod *method, MonoDomain *domain, g
 			mono_error_cleanup (error);
 			return ERR_INVALID_ARGUMENT;
 		}
-		if (!mono_verifier_is_method_valid_generic_instantiation (inflated))
```
Comment (vargaz) [inline src/mono/mono/mini/debugger-agent.c:8514]: This, and probably some other places check whenever what the app wants to construct is valid, it shouldn't be removed.

File: src/mono/mono/metadata/class-init.c [hunk 2]
```diff
@@ -2950,9 +2944,6 @@ mono_class_init_internal (MonoClass *klass)
 
 	mono_class_setup_interface_offsets_internal (klass, first_iface_slot, TRUE);
 
-	if (mono_class_is_ginst (klass) && !mono_verifier_class_is_valid_generic_instantiation (klass))
```
Comment (vargaz) [inline src/mono/mono/metadata/class-init.c:2953]: This one needs to stay as well.

File: src/mono/mono/metadata/icall.c [hunk 3]
```diff
@@ -3086,14 +3085,6 @@ ves_icall_RuntimeType_MakeGenericType (MonoReflectionTypeHandle reftype, MonoArr
 		return MONO_HANDLE_CAST (MonoReflectionType, NULL_HANDLE);
 	}
 
-	MonoClass *klass = mono_class_from_mono_type_internal (geninst);
-
-	/*we might inflate to the GTD*/
-	if (mono_class_is_ginst (klass) && !mono_verifier_class_is_valid_generic_instantiation (klass)) {
-		mono_error_set_argument (error, "typeArguments", "Invalid generic arguments");
```
Comment (vargaz) [inline src/mono/mono/metadata/icall.c:3093]: Same.

File: src/mono/mono/metadata/loader.c [hunk 4]
```diff
@@ -787,14 +780,6 @@ mono_method_get_signature_checked (MonoMethod *method, MonoImage *image, guint32
 
 			sig = (MonoMethodSignature *)cache_memberref_sig (image, sig_idx, sig);
 		}
-		/* FIXME: we probably should verify signature compat in the dynamic case too*/
-		if (!mono_verifier_is_sig_compatible (image, method, sig)) {
-			guint32 klass = cols [MONO_MEMBERREF_CLASS] & MONO_MEMBERREF_PARENT_MASK;
```
Comment (vargaz) [inline src/mono/mono/metadata/loader.c:792]: This one needs to stay too, it checks that the caller and callee assembly agree on the signature.

File: src/mono/mono/mini/debugger-agent.c (hunk 1 above)
Comment (lambdageek) [reply src/mono/mono/mini/debugger-agent.c:8514]: It always returns TRUE in netcore (with verified disabled)

**Comment by lambdageek** [comment]
Ah no, you're right @vargaz, I deleted too much code. :-(

File: src/mono/mono/mini/debugger-agent.c (hunk 1 above)
Comment (lambdageek) [reply src/mono/mono/mini/debugger-agent.c:8514]: no I'm wrong.  Need to rework this.

**Pull request closed without merging**